### ISSUE TITLE
Which Windows installer is recommended?

### ISSUE THREAD

**Issue by velle** [qgis/QGIS-Website#374]
### URL

https://qgis.org/download/

### Type(s) of Problems

Other

### Summary

# What is a "regular user"?

From https://qgis.org/download/, under "Download for Windows". 

> The OSGeo4W installer is recommended for regular users or organization deployments. It allows to have several QGIS versions in one place, and to keep each component up-to-date individually without having to download the whole package.

Please note that English is not my native language.  

So what is a "regular user"? Initially I thought it meant a "normal" or "typical" user. But if so, the text suggests basically everyone to download the OSGeo4W installer, right? At least I don't see who should download any of the other installers?

It took me a few minutes to realize that perhaps "regular" means someone who uses it regularly? Is that what is meant?

If I have understood the intention right, then "regular" is not the right term to be used here. Even someone who regularly uses QGIS is not necessarily someone who will need to run various versions in parallel or who will have the super user-ish skills to use the OSGeo4 installer. 

# OSGeo4W installer vs OSGeo4W network installer

The text refers to "OSGeo4W installer" but the download button reads "OSGeo4W network installer". Is that the same thing? I think so. 

# My suggestion

Something along the text below. 

- LTR 3.34: Recommended. Long Term Release. This version has been carefully tried and tested, and is considered stable and free of significant bugs.   
- 3.38: Most recent official release, but less stable than the above LTR version. 
- OSGeo4W Network Installer: Allows installing various versions concurrently, allows installing without admin rights, allows custom selection of modules, allows upgrading modules individually, makes future upgrades / installations faster and easier. Suitable for users with a bit  more experience with QGIS and IT in general.

**Comment by m-kuhn** [comment]
Thanks for raising this @velle, I was looking at the page right now and had similar thoughts.

Some more notes:

> This page provides binary packages (installers).

Stating the obvious / self-evident, I think this sentence could be removed without losing value.

> The current version is QGIS 3.40.2 'Bratislava' and was released on 2024-12-20.
> 
> The long-term builds currently offer QGIS 3.34.14 'Prizren'.
> 
> QGIS is available on Windows, macOS, Linux, Android and iOS.

Is already "advanced" information, should go below the download links

I think we should have one "main" download link which will be more prominent, so anyone without specific needs clicks on it. This can be latest LTR, with a smallprint below as suggested by @velle . It would be good to also have a "title" for that, so we can explicitly refer to it elsewhere (see https://github.com/qgis/QGIS-Website/pull/522#discussion_r1918070597), e.g. "Standalone QGIS long term version installer".

And then we can have two more links for "Standalone QGIS latest version installer" and "OSGeo4W Package Manager" (or network installer or whatever, as long as it's consistent :-) )

![image](https://github.com/user-attachments/assets/d47b8de3-6cd7-41f9-a399-1687cb739979)

@timlinux do you have any thoughts / comments on this?

**Comment by m-kuhn** [comment]
Information by @agiudiceandrea 

there was a terminology before, maybe we can recover it. At least partially (and only the terminology and not the complete page :-) )

![Image](https://github.com/user-attachments/assets/78f6decb-c129-4dd5-a7e6-f5cd5adc6176)

**Comment by jef-n** [comment]
The OSGeo4W installer is what I recommend.   The MSIs is just for people who prefer standalone installers or require MSIs for some reason (and in turn have to download a huge blob containing a relatively small new QGIS with mostly unchanged set of dependencies on each update. But they are also handy if you for some reason need an older version.

**Comment by velle** [comment]
I found this https://gis.stackexchange.com/questions/455371/difference-between-qgis-download-and-osgeow4-network-installer. 

> In the past, whenever a new version of QGIS was released, I downloaded the MSI file from the QGIS website. Once the installer was run, it would install the new version as if it was the first installation of the software. I would eventually end up with 4 or 5 versions on my system concurrently, each taking up 2-3 GB of hard drive space.

Is that really how the MSI installers work? If so, isnt that different from how installers normally work and what users would expect?

**Comment by timlinux** [comment]
> Thanks for raising this [@velle](https://github.com/velle), I was looking at the page right now and had similar thoughts.
> 
> Some more notes:
> 
> > This page provides binary packages (installers).
> 
> Stating the obvious / self-evident, I think this sentence could be removed without losing value.
> 
> > The current version is QGIS 3.40.2 'Bratislava' and was released on 2024-12-20.
> > The long-term builds currently offer QGIS 3.34.14 'Prizren'.
> > QGIS is available on Windows, macOS, Linux, Android and iOS.
> 

See also the edits I proposed here: https://github.com/qgis/QGIS-Website/pull/522#issuecomment-2595761696

> Is already "advanced" information, should go below the download links
> 
> I think we should have one "main" download link which will be more prominent, so anyone without specific needs clicks on it. This can be latest LTR, with a smallprint below as suggested by [@velle](https://github.com/velle) .

👍🏽

> It would be good to also have a "title" for that, so we can explicitly refer to it elsewhere (see [#522 (comment)](https://github.com/qgis/QGIS-Website/pull/522#discussion_r1918070597)), e.g. "Standalone QGIS long term version installer".
> 
> And then we can have two more links for "Standalone QGIS latest version installer" and "OSGeo4W Package Manager" (or network installer or whatever, as long as it's consistent :-) )
> 
> ![image](https://github.com/user-attachments/assets/d47b8de3-6cd7-41f9-a399-1687cb739979)
> 
> [@timlinux](https://github.com/timlinux) do you have any thoughts / comments on this?


In https://github.com/qgis/QGIS-Website/pull/522#issuecomment-2595761696 I proposed to change the intro text to this:

This page provides binary packages (installers) for QGIS.
The current version is QGIS 3.40.2 'Bratislava' which was released on 2024-12-20.
The long-term builds currently provide QGIS 3.34.14 'Prizren'. Long Term Release (LTR) builds are intended for those who value stability over having the latest features. If you are unsure which version is best for you, download the LTR,
QGIS is available on Windows, macOS, Linux, Android and iOS.

For the actual download section we could then organise it like this:

Online installer:

[ OSGeo4W Network Installer ]

This installer is the best way to keep QGIS up to date, run multiple versions on your system and keep the load on our download servers to a minimum. Using this installer, you can fetch updates to QGIS efficiently since it only downloads the changed sub packages. The installer provides an interactive menu where you can choose which components you wish to install, including special drivers like ECW support etc. During the installation process, you can choose whether you want the Long Term Release (LTR)  version or the latest version of QGIS (or both). We recommend the LTR version as it has undergone more quality assurance than the Latest Version. The latter is a good choice if you need the newest features and are not as concerned about stability.

Offline installers:

[ Long Term Version for Windows (3.34 LTR) ]
[ Latest Version for Windows (3.40) ]

These installers are for users who wish to easily share the download e.g. putting it on a USB key or network share. The  QGIS application and its component libraries need to be completely downloaded each time there is a new release that you want to upgrade to. We recommend the LTR version as it has undergone more quality assurance than the Latest Version. The latter is a good choice if you need the newest features and are not as concerned about stability.


⚠️ Since QGIS 3.20 we only ship 64-bit Windows executables.

**Comment by velle** [comment]
@timlinux Regarding the way you divided it into "Online installer" and "Offline installers". It is possible to use OSGeo4W to do offline installations; and if I understand correctly it is often the best choice for doing so (see https://trac.osgeo.org/osgeo4w/wiki/FAQ#HowdoIperformanofflineorcomputerlabinstall). How about just "OSGeo4W Installer" and "MSI Installers".

**Comment by velle** [comment]
I have realized that these two pages are very closely related:
- https://qgis.org/download/
- https://www.qgis.org/resources/installation-guide

To avoid too much redundancy, would it be possible to keep the text on the Download page at a bare minimum, maybe 1-2 for each of the three options, and then refer to the Installation Guide for more information? 

And then perhaps we should have a look at the Installation Guide page, and see if the discussions / suggestions from this thread can be used there instead.

**Comment by timlinux** [comment]
> [@timlinux](https://github.com/timlinux) Regarding the way you divided it into "Online installer" and "Offline installers". It is possible to use OSGeo4W to do offline installations; and if I understand correctly it is often the best choice for doing so (see https://trac.osgeo.org/osgeo4w/wiki/FAQ#HowdoIperformanofflineorcomputerlabinstall). How about just "OSGeo4W Installer" and "MSI Installers".

Hi. Thanks for your message. Yes I am aware of the possibility to use the OSGEO4W installer offline but I think it is good to cater for 'mainstream' use cases. The terms OSGEO4W Installer and MSI installer are language that is not very accessible to every day users, and so I would prefer it if we used simple, clear language as much as possible.

**Comment by agiudiceandrea** [comment]
> Information by [@agiudiceandrea](https://github.com/agiudiceandrea)
> 
> there was a terminology before, maybe we can recover it. At least partially (and only the terminology and not the complete page :-) )
> 
> ![Image](https://github.com/user-attachments/assets/78f6decb-c129-4dd5-a7e6-f5cd5adc6176)

The term "Standalone installer" (and "OSGeo4W installer") is also currently used in the page at https://www.qgis.org/resources/installation-guide/#standalone-installers

![Image](https://github.com/user-attachments/assets/609d7c90-5bd1-4f59-855c-94330d8d28ce)

**Comment by Xpirix** [comment]
Hi everyone,

I've included for you the proposed PR at #530 . I tried to take into consideration everyone's suggestions and merged them. Please feel free to share your feedback there.

**Comment by velle** [comment]
@Xpirix You did a good job :) 

I think it would be appropriate to close this issue now, and go with @Xpirix 's PR, which is a great improvement. 

There are some details regarding the installers that I am unsure of, and I suspect others are unsure (or misinformed) as well. It would be great to improve the Installation Guide such that those details are clear to everyone who reads it. But I think such details could go in a separate issue focused on that, if/when someone decides to create such an issue. 

Examples of technical details which I'm unsure of:
- I believe that when downloading and installing using an MSI, OSGeo4W Installer is actually also installed locally. If I'm right, the MSI version is just as good as the other for continous future upgrades.
- Is OSGeo4W really better for running different versions concurrently? And if so, in which way? I'm not sure its better. 
- When installing with MSI, I think previous versions are not deleted/overridden. Is that really so? What if I upgrade using OSGeo4W installer, will that override previous versions? What about previous versions that were not installed with OSGeo4W, but with MSI installer?

**Comment by jef-n** [comment]
>     * I believe that when downloading and installing using an MSI, OSGeo4W Installer is actually also installed locally. If I'm right, the MSI version is just as good as the other for continous future upgrades.

Except that you don't need the MSI in the first place.  You could start with OSGeo4W installer directly.

The default installation directory of the MSI reflects the included QGIS version, so once that is upgraded with the OSGeo4W that might become misleading.

>     * Is OSGeo4W really better for running different versions concurrently? And if so, in which way? I'm not sure its better.

Well, you also get a OSGeo4W install if you use the MSI (in a separate directory reflecting the QGIS version).   So you could use the included OSGeo4W installer also to install different available QGIS versions available in OSGeo4W at that point - of upgrade to the current ones, if the MSI is older. But if you wanted that, why would you start with the MSI to begin with?

The QGIS' within one OSGeo4W installation directory share their dependencies (ie. proj, gdal, Qt…)
 
>     * When installing with MSI, I think previous versions are not deleted/overridden. Is that really so? What if I upgrade using OSGeo4W installer, will that override previous versions? What about previous versions that were not installed with OSGeo4W, but with MSI installer?

Yes.  You get an  independent install in a separate directory.  And you'd upgrade  only that directory if you use the included OSGeo4W installer (unless you explicitly tell it to look elsewhere).
